File: app/src/main/java/com/innovation/rain/feature/agentlogin/injection/AgentLoginBuilderModule.java
package com.innovation.rain.feature.agentlogin.injection;

import com.sf0404.core.application.scope.PerView;
import com.innovation.rain.feature.agentlogin.view.AgentLoginFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;


@Module
public abstract class AgentLoginBuilderModule {
    @PerView
    @ContributesAndroidInjector(modules = {AgentLoginViewModule.class
//            , AgentLoginUseCaseModule.class //TODO
    })
    abstract AgentLoginFragment contributeFragment();
}

File: app/src/main/java/com/innovation/rain/feature/agentlogin/injection/AgentLoginBuilderModule.kt
package com.innovation.rain.feature.agentlogin.injection

import com.innovation.rain.feature.agentlogin.view.AgentLoginFragment
import com.sf0404.core.application.scope.PerView
import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Binds all sub-components within the app.
 */
@Module
abstract class AgentLoginBuilderModule {
    @PerView
    @ContributesAndroidInjector(modules = [AgentLoginViewModule::class, AgentLoginUseCaseModule::class])
    abstract fun contributeFragment(): AgentLoginFragment
}